### PR TITLE
Move libs from test-infra to pkg

### PR DESCRIPTION
**Pull request by chizhg**
There are still some libs in test-infra that are vendored by serving. Move them to pkg/test.

/cc @chaodaiG

### CODE REVIEW

File: test/performance/performance.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
+/*
+Copyright 2019 The Knative Authors
```
Comment (mattmoor) [inline test/performance/performance.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/testgrid/testgrid.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2019 The Knative Authors
```
Comment (mattmoor) [inline test/testgrid/testgrid.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/testgrid/testgrid_test.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2019 The Knative Authors
```
Comment (mattmoor) [inline test/testgrid/testgrid_test.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/testgrid/yaml.go [hunk 4]
```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2019 The Knative Authors
```
Comment (mattmoor) [inline test/testgrid/yaml.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/testgrid/testgrid_test.go [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+package testgrid
+
+import (
+	"io/ioutil"
+	"os"
+	"path"
+	"testing"
+
+	"knative.dev/pkg/test/junit"
```
Comment (mattmoor) [inline test/testgrid/testgrid_test.go]: Format Go code:
```suggestion

```

File: test/testgrid/testgrid_test.go [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+	"knative.dev/pkg/prow"
```
Comment (mattmoor) [inline test/testgrid/testgrid_test.go]: Format Go code:
```suggestion
	"knative.dev/pkg/prow"
	"knative.dev/pkg/test/junit"
```

File: test/testgrid/testgrid_test.go (hunk 3 above)
Comment (mattmoor) [inline test/testgrid/testgrid_test.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

Comment (mattmoor) [inline test/testgrid/testgrid_test.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: test/testgrid/testgrid.go [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+	"knative.dev/pkg/test/helpers"
+	"knative.dev/pkg/test/junit"
+	"knative.dev/pkg/test/prow"
+)
+
+const (
+	filePrefix = "junit_"
+	extension  = ".xml"
+)
+
```
Comment (chaodaiG) [inline test/testgrid/testgrid.go]: I see that you removed some of the stuff, like the jobname : tabname map, are they intended to be kept in test-infra? 

https://github.com/knative/test-infra/blob/0042e9ca752c8339c3c245e1d2c76b891a144b7e/pkg/testgrid/testgrid.go#L40

Comment (chizhg) [reply test/testgrid/testgrid.go]: Yes, they should still be kept in test-infra as they are tied to the testgrid.yaml file that lives in test-infra repo.

**Pull request merged**